commons/src/utils/permissions.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-FileCopyrightText: 2023 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { NotePermissions, SpecialGroup } from './permissions.types.js'
+
+export const userIsOwner = (
+  permissions: NotePermissions,
+  user?: string
+): boolean => {
+  return !!user && permissions.owner === user
+}
+export const userCanEdit = (permissions: NotePermissions, user?: string) => {
+  const isOwner = userIsOwner(permissions, user)
+  const mayWriteViaUserPermission = permissions.sharedToUsers.some(
+    (value) => value.canEdit && value.username === user
+  )
+  const mayWriteViaGroupPermission = permissions.sharedToGroups.some(
+    (value) =>
+      value.groupName === SpecialGroup.LOGGED_IN && value.canEdit && !!user
+  )
```

Comment:
This checks for `!!user` in every iteration of the loop that runs over the sharedToGroups array. You could change this to the following:

```suggestion
  const mayWriteViaGroupPermission = !!user && permissions.sharedToGroups.some(
    (value) =>
      value.groupName === SpecialGroup.LOGGED_IN && value.canEdit
  )
```